vnext/Microsoft.ReactNative/ReactApplication.cpp

```diff
@@ -103,6 +103,14 @@ void ReactApplication::OnActivated(Windows::ApplicationModel::Activation::IActiv
   if (e.Kind() == Windows::ApplicationModel::Activation::ActivationKind::Protocol) {
     auto protocolActivatedEventArgs{e.as<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs>()};
     react::uwp::LinkingManagerModule::OpenUri(protocolActivatedEventArgs.Uri());
+  } else if (e.Kind() == Windows::ApplicationModel::Activation::ActivationKind::ToastNotification) {
+    auto toastActivatedEventArgs{e.as<Windows::ApplicationModel::Activation::ToastNotificationActivatedEventArgs>()};
+    try {
+      Windows::Foundation::Uri u{toastActivatedEventArgs.Argument()};
```

Comment:
The toast event args are free to be any string so requiring it to be a Uri is a convention. I wonder if we should encode the argument as a fixed but parametrized Uri instead.
For example: `toast://encodedArgument` then the app can extract the argument from the Uri.

Reply:
I thought along similar lines. I'm happy to change it to whatever, but the `authority` section of a URI does seem a bit odd. Other options are to do `toast:///?encodedArgument` or `ms-appdata:///?encodedArgument`.

I think you'd still be requiring the toast author to format the string in a certain way or to deal with encoding on the js event handler side in a kind of weird way so my instinct was to just allow the toast author and js handler author both use URIs that aren't modified by react-native-windows.

Like I said, I'm happy to implement something like you're suggesting, just let me know what that is.

Reply:
Looking at this, why does the LinkingManagerModule force us to use a Uri object at all.  Its just a string as far as RN is concerned.  Shouldn't we just change LinkingManagerModule::OpenUri to take a string and avoid the extra object conversions?

Reply:
@ryfow good points. Let me mull this over but I'd be ok with toast:///?encodedArgument too.
I think we will want to follow the format described here: https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/send-local-toast
which is "QueryString" syntax. The problem is linking expects a Uri not just a string so we have to bridge the two worlds. `toast:///?queryString` would do that

@acoates-ms well linking comes from core RN though, maybe we just need a new API on linking https://reactnative.dev/docs/linking.html

Reply:
That "Send a local toast notification" doc is saying that the toast author should format the argument as a query string. Your plan as I understand it would just concatenate that (we assume) encoded qs onto `toast:///?`. We're kind of back in the same place we started with trying to create a URI out of that string that may or may not be a valid URI.

We could URL encode again in `OnActivated`, but that would require double decoding on the JS side.

Reply:
I pushed a `toast:///?queryString` implementation. Uri seems pretty flexible about what you can put in the query string but the encoding should still be done in the code creating the Toast.